pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -361,7 +361,15 @@ func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType cluster
 	}
 	ec := NewMutableCluster(c)
 	switch discoveryType {
-	case cluster.Cluster_STRICT_DNS, cluster.Cluster_LOGICAL_DNS:
+	case cluster.Cluster_LOGICAL_DNS:
+		if len(localityLbEndpoints) != 1 || len(localityLbEndpoints[0].LbEndpoints) != 1 {
+			log.Warnf("%s cluster of type DNS_ROUND_ROBIN should have a single locality_lb_endpoint and a single lb_endpoint."+
+				" This can happen if two Service Entries are created with same host name, resolution as DNS_ROUND_ROBIN and with"+
+				" same/different endpoints.", name)
+			return nil
```

Comment:
So we just ignore it entirely? Should we be choosing one of them?

Reply:
I think which one to choose is little difficult - I mean which endpoint to choose. If it is same we could remove the duplicate but it is different we do not know which one to choose. We could always choose the first one and send but I thought dropping it would make the problem obvious.

In addition to this, I am wondering if we can add an analyzer check also?

Reply:
It just seems like we may have:

* SE with 1 host, it works fine
* Add a new SE, suddenly everything breaks

Typically we resolve these conflicts with "oldest one wins", but that may be a bit hard to enforce in this part of the code. But having 0 seems a bit harmful

Reply:
Correct, but sending the second one (new one) also has undesired side affects. Not sure what is the best way here. Any suggestions?  Other way I thought is, restrict at the time adding it to store - but since we do not key by host, it becomes complicated there.

Any other ideas?

Reply:
What about in buildLocalityLbEndpoints? We have access to the Service.CreationTimestamp when looping over `instances`? Maybe we can filter there?

Reply:
Let me try there